framework/include/variables/MooseVariableData.h

```diff
@@ -478,6 +478,11 @@ class MooseVariableData
     return _matrix_tag_u[tag];
   }
 
+  /**
+   * The solution states that are needed for this variable data.
+   */
```

Comment:
This comment is not clear to me. It says `states` but only asking for one single returned number. Does `0` mean current solution only, `1` mean current plus old, etc.?

Reply:
See https://github.com/idaholab/moose/pull/16363#discussion_r551435479. 

I plan on making this more clear.

Reply:
Oh, the one in `MooseVariableFieldBase.h` seems clearer and should be copied here.